### PR TITLE
Do not update cached policies when receiving tombstoned policies

### PR DESCRIPTION
**Pull request by xichen2020**
cc @cw9 @jeromefroe 

This PR fix an issue when ingesting metrics with tombstoned policies. Previously if the policies indicate the metric is tombstoned, the internal policies are updated and cached, and the entry is marked as tombstoned. This causes an issue when there are two different raw metrics matching the same rollup rule, one of which matches the rule and the other does not. As a result, the agg tier will receive a tombstoned policy for the second raw metric, mark the entry tombstoned, even though the entry is still very much alive due to the first raw metric still matching that rule and generating datapoints for that entry.

The fix is simply to not update the internal cached policies when receiving a tombstoned policy and instead rely on background ticking goroutine to sweep entries that have not been recently accessed.

### CODE REVIEW

File: aggregator/entry_test.go [hunk 1]
```diff
@@ -807,9 +802,13 @@ func testEntryAddMetricWithPoliciesList(
 				id := elem.Value.(*CounterElem).ID()
 				require.Equal(t, testCounterID, id)
 				aggregations := elem.Value.(*CounterElem).values
-				require.Equal(t, 1, len(aggregations))
-				require.Equal(t, alignedStart.UnixNano(), aggregations[0].timeNanos)
-				require.Equal(t, int64(1234), aggregations[0].counter.Sum())
+				if !expectedShouldAdd {
```
Comment (cw9) [inline aggregator/entry_test.go]: nit: 
`if expectedShouldAdd {...} else {...}`

File: aggregator/entry_test.go [hunk 2]
```diff
@@ -818,9 +817,13 @@ func testEntryAddMetricWithPoliciesList(
 				id := elem.Value.(*TimerElem).ID()
 				require.Equal(t, testBatchTimerID, id)
 				aggregations := elem.Value.(*TimerElem).values
-				require.Equal(t, 1, len(aggregations))
-				require.Equal(t, alignedStart.UnixNano(), aggregations[0].timeNanos)
-				require.Equal(t, 18.0, aggregations[0].timer.Sum())
+				if !expectedShouldAdd {
```
Comment (cw9) [inline aggregator/entry_test.go]: same

File: aggregator/entry_test.go (hunk 1 above)
Comment (xichen2020) [reply aggregator/entry_test.go]: Meh I usually put the shorter block first to make it easier on the eyes.

File: aggregator/entry_test.go (hunk 2 above)
Comment (xichen2020) [reply aggregator/entry_test.go]: Same as above.

**Pull request merged**